.github/workflows/ci.yml

```diff
@@ -56,3 +56,11 @@ jobs:
     - name: Build
       shell: bash
       run: cmake --build $GITHUB_WORKSPACE/build --config Release --target install
+
+    - name: Check formatting
+      if: matrix.platform.name == 'Linux Clang'
+      run: |
+        sudo apt-get install clang-format
+        cmake --build $GITHUB_WORKSPACE/build --target format
+        git diff
+        exit `git status --porcelain | wc -l`
```

Comment:
This could be a separate job, with the benefit of showing success or failure of formatting only, instead or combined build and formatting. Something like (untested):
```
  format:
    name: Check formatting
    runs-on: ubuntu-latest

    steps:
    - name: Checkout Code
      uses: actions/checkout@v2

    - name: Install Dependencies
      run: sudo apt-get update && sudo apt-get install clang-format

    - name: Configure CMake
      run: cmake -S $GITHUB_WORKSPACE -B $GITHUB_WORKSPACE/build -DCMAKE_VERBOSE_MAKEFILE=ON
      
    - name: Run clang-format
      run: |
        cmake --build $GITHUB_WORKSPACE/build --target format
        git diff
        exit `git status --porcelain | wc -l`
```

Reply:
Yep was working on that, just wanted to test it out quick. still getting my head around github actions 😅